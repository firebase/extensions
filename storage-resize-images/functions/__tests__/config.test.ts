import mockedEnv from "mocked-env";

const environment = {
  LOCATION: "us-central1",
  IMG_BUCKET: "extensions-testing.appspot.com",
  CACHE_CONTROL_HEADER: undefined,
  IMG_SIZES: "200x200",
  RESIZED_IMAGES_PATH: undefined,
  DELETE_ORIGINAL_FILE: "true",
};

let restoreEnv;

const { config, deleteImage } = global;
let deleteTypeCounter = 0;

describe("extension", () => {
  beforeEach(() => {
    jest.resetModules();
    if (deleteTypeCounter === 0) {
      restoreEnv = mockedEnv(environment);
    } else if (deleteTypeCounter === 1) {
      restoreEnv = mockedEnv({ ...environment, DELETE_ORIGINAL_FILE: "false" });
    } else if (deleteTypeCounter === 2) {
      restoreEnv = mockedEnv({
        ...environment,
        DELETE_ORIGINAL_FILE: "on_success",
      });
    }
  });

  afterEach(() => restoreEnv());

  test("configuration detected from environment variables", async () => {
    const mockConfig = config();

    expect(mockConfig).toMatchSnapshot({});
  });

  test("always delete original file", async () => {
    const mockConfig = config();
    const mockDeleteImage = deleteImage();
    deleteTypeCounter++;
    expect(mockConfig.deleteOriginalFile).toEqual(mockDeleteImage.always);
  });

  test("never delete original file", async () => {
    const mockConfig = config();
    const mockDeleteImage = deleteImage();
    deleteTypeCounter++;
    expect(mockConfig.deleteOriginalFile).toEqual(mockDeleteImage.never);
  });
  test("delete original file on success", async () => {
    const mockConfig = config();
    const mockDeleteImage = deleteImage();

    expect(mockConfig.deleteOriginalFile).toEqual(mockDeleteImage.onSuccess);
  });
});
