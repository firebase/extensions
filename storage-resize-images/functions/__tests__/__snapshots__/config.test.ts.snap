// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extension configuration detected from environment variables 1`] = `
Object {
  "bucket": "extensions-testing.appspot.com",
  "cacheControlHeader": undefined,
  "deleteOriginalFile": 0,
  "excludePathList": undefined,
  "imageSizes": Array [
    "200x200",
  ],
  "imageTypes": undefined,
  "includePathList": undefined,
  "resizedImagesPath": undefined,
}
`;
